syntax = "proto3";

package NCloud.NBlockStore.NProto;

option java_package = "ru.yandex.cloud.blockstore.proto";
option go_package = "a.yandex-team.ru/cloud/blockstore/public/api/protos";

////////////////////////////////////////////////////////////////////////////////
// Encryption mode

enum EEncryptionMode
{
    NO_ENCRYPTION = 0;
    ENCRYPTION_AES_XTS = 1;
}

////////////////////////////////////////////////////////////////////////////////
// Encryption key path

message TKeyPath
{
    oneof Path
    {
        // Identifier of the keyring containing the key.
        uint32 KeyringId = 1;

        // Path to the file containing the key.
        bytes FilePath = 2;
    }
}

////////////////////////////////////////////////////////////////////////////////
// Encryption spec

message TEncryptionSpec
{
    // Encryption mode.
    EEncryptionMode Mode = 1;

    oneof KeyParam
    {
        // Encryption key path.
        TKeyPath KeyPath = 2;

        // Encryption key hash for snapshot.
        bytes KeyHash = 3;
    }
}
